### ISSUE TITLE
Compilation error: undefined local variable or method 'future' for Logger

### ISSUE THREAD

**Issue by bogga** [cooperhammond/irs#63]
Running the install from source instructions exactly on Ubuntu 20.04.1 LTS (WSL 2), I get `Error: undefined local variable or method 'future' for Logger`.

This is the full stack trace:
```
$ shards build
Dependencies are satisfied
Building: irs
Error target irs failed to compile:
In src/glue/mapper.cr:4:8

 4 | JSON.mapping(
          ^------
Warning: Deprecated JSON.mapping. use JSON::Serializable instead (the legacy behaviour is also available in a shard at github:crystal-lang/json_mapping.cr)

In src/glue/mapper.cr:4:3

 4 | JSON.mapping(
     ^
Warning: expanding macro


There was a problem expanding macro 'mapping'

Called macro defined in /usr/share/crystal/src/json/mapping.cr:232:3

 232 | macro mapping(**_properties_)

Which expanded to:

 > 1 |     ::JSON.mapping({tracks: {type: PlaylistTracksMapper, setter: true}, id: String, images: JSON::Any, name: String, owner: JSON::Any, type: String})
   2 |
Warning: Deprecated JSON.mapping. use JSON::Serializable instead (the legacy behaviour is also available in a shard at github:crystal-lang/json_mapping.cr)

In src/glue/mapper.cr:18:8

 18 | JSON.mapping(
           ^------
Warning: Deprecated JSON.mapping. use JSON::Serializable instead (the legacy behaviour is also available in a shard at github:crystal-lang/json_mapping.cr)

In src/glue/mapper.cr:28:8

 28 | JSON.mapping(
           ^------
Warning: Deprecated JSON.mapping. use JSON::Serializable instead (the legacy behaviour is also available in a shard at github:crystal-lang/json_mapping.cr)

In src/glue/album.cr:12:7

 12 | def find_it
          ^------
Warning: this method overrides SpotifyList#find_it() which has an explicit return type of JSON::Any.

Please add an explicit return type (JSON::Any or a subtype of it) to this method as well.

The above warning will become an error in a future Crystal version.

In src/glue/playlist.cr:16:7

 16 | def find_it
          ^------
Warning: this method overrides SpotifyList#find_it() which has an explicit return type of JSON::Any.

Please add an explicit return type (JSON::Any or a subtype of it) to this method as well.

The above warning will become an error in a future Crystal version.

A total of 6 warnings were found.
Showing last frame. Use --error-trace for full trace.

In src/interact/logger.cr:41:14

 41 | called = future {
               ^-----
Error: undefined local variable or method 'future' for Logger
```

This is my crystal version, if it helps
```
$ crystal --version
Crystal 0.35.1 [5999ae29b] (2020-06-19)

LLVM: 8.0.0
Default target: x86_64-unknown-linux-gnu
```

**Comment by bogga** [comment]
Full trace:
```
$ shards build --error-trace
Dependencies are satisfied
Building: irs
Error target irs failed to compile:
In src/glue/mapper.cr:4:8

 4 | JSON.mapping(
          ^------
Warning: Deprecated JSON.mapping. use JSON::Serializable instead (the legacy behaviour is also available in a shard at github:crystal-lang/json_mapping.cr)

In src/glue/mapper.cr:4:3

 4 | JSON.mapping(
     ^
Warning: expanding macro


There was a problem expanding macro 'mapping'

Called macro defined in /usr/share/crystal/src/json/mapping.cr:232:3

 232 | macro mapping(**_properties_)

Which expanded to:

 > 1 |     ::JSON.mapping({tracks: {type: PlaylistTracksMapper, setter: true}, id: String, images: JSON::Any, name: String, owner: JSON::Any, type: String})
   2 |
Warning: Deprecated JSON.mapping. use JSON::Serializable instead (the legacy behaviour is also available in a shard at github:crystal-lang/json_mapping.cr)

In src/glue/mapper.cr:18:8

 18 | JSON.mapping(
           ^------
Warning: Deprecated JSON.mapping. use JSON::Serializable instead (the legacy behaviour is also available in a shard at github:crystal-lang/json_mapping.cr)

In src/glue/mapper.cr:28:8

 28 | JSON.mapping(
           ^------
Warning: Deprecated JSON.mapping. use JSON::Serializable instead (the legacy behaviour is also available in a shard at github:crystal-lang/json_mapping.cr)

In src/glue/album.cr:12:7

 12 | def find_it
          ^------
Warning: this method overrides SpotifyList#find_it() which has an explicit return type of JSON::Any.

Please add an explicit return type (JSON::Any or a subtype of it) to this method as well.

The above warning will become an error in a future Crystal version.

In src/glue/playlist.cr:16:7

 16 | def find_it
          ^------
Warning: this method overrides SpotifyList#find_it() which has an explicit return type of JSON::Any.

Please add an explicit return type (JSON::Any or a subtype of it) to this method as well.

The above warning will become an error in a future Crystal version.

A total of 6 warnings were found.
In src/irs.cr:8:1

 8 | main()
     ^---
Error: instantiating 'main()'


In src/irs.cr:5:7

 5 | cli.act_on_args
         ^----------
Error: instantiating 'CLI#act_on_args()'


In src/bottle/cli.cr:85:9

 85 | s.grab_it
        ^------
Error: instantiating 'Song#grab_it()'


In src/glue/song.cr:94:12

 94 | Ripper.download_mp3(url.as(String), @filename)
             ^-----------
Error: instantiating 'Ripper:Module#download_mp3(String, String)'


In src/interact/ripper.cr:41:14

 41 | return l.start do |line, index|
               ^----
Error: instantiating 'Logger#start()'


In src/interact/logger.cr:41:14

 41 | called = future {
               ^-----
Error: undefined local variable or method 'future' for Logger
```

**Comment by yyyyyyyan** [comment]
I got the same thing. I installed crystal from Arch Linux repo (pacman -S crystal shards).

**Comment by NoctisAtrae** [comment]
Up +1 ! I have the same issues !

**Comment by cooperhammond** [comment]
Should be fixed with pull #64, the .35 update broke it

**Comment by bogga** [comment]
confirmed: it builds! (needed to install `libxml2-dev` and `libyaml-dev` first, just in case anyone needed that help)
